### ISSUE TITLE
[Bug]: Some mechanoid ammo cannot be sold.

### ISSUE THREAD

**Issue by N7Huntsman** [CombatExtended-Continued/CombatExtended#7]
**Complete the template below. Reports that do not follow the template will not be deleted.**
**Title should be:** [Bug]: (Short description of bug)

**Specifications**

Rimworld version: 1.1
Combat Extended version: 1.1.20
Combat Extended source (Steam, GitHub, etc.): Steam
Your operating system: Windows 10 x64

**Description**
Despite having a market value iassigned to them and the appropriate trade tags, some ammo dropped by mechanoids does not display a market value field in the stats menu and so it can't be sold to trader. This happens with the 5x35mm charged cartridge (charge lance) and the 12x64mm charged (heavy change blaster), but does not happen with the 80x256mm fuel cell (inferno cannon).

**Expected behavior**
Expected to be able to sell mechanoid ammo to traders.

**To reproduce**
1. Spawn affect mechanoid ammo.
2. Inspect in stat menu and see that it has no market value field.

**Screenshots & log dumps**
n/a

**Complete the following checklist**
I hereby verify that I have done the following:
[x] Confirmed that my game version and load order are correct.
[x] Confirmed that I am running the appropriate and most updated version of Combat Extended and required compatibility patches.
[x] Confirmed I am not running any mods with known incompatibilities with Combat Extended.
[x] Disabled Combat Extended and attempted to reproduce the behavior without success.

**Comment by LockdownX7** [comment]
There's two factors going into this:

First is that vanilla removed the destroyOnDrop tag for Inferno cannons in the 1.1 update. I assume that's not an intentional change, unless Ludeon added a new way for colonists to obtain and use that weapon. Otherwise, CE should patch that tag back in, same as it does with the charge lance.

Second is this bit of code in AmmoInjector:
```
if (!ammoDef.Users                                                                          //If there exists NO gun..
	.Any(x => !x.destroyOnDrop                                                              //.. which DOESN'T destroy on drop (e.g all guns destroy on drop)
				|| (x.weaponTags != null && x.weaponTags.Contains("TurretGun")              //.. or IS part of a Turret..
					&& DefDatabase<ThingDef>.AllDefs.Where(y => y.building?.turretGunDef == x)                //.. as long as ALL turrets using the gun are non-mechcluster turrets
						.All(y => !y.building?.buildingTags?.Contains(MechClusterGenerator.MechClusterMemberTag) ?? true))))                                                                  
	continue;                                                                               //Then this ammo's tradeability and craftability are ignored
```

Seems the intention is to prevent some ammo types from mech-cluster turrets from ever having tradeability or craftability, but it's inadvertently covering all mechanoid weapon ammo as well. Because of the vanilla change mentioned above, 80x256mm is the one exception, because it's used by the Inferno cannon so !x.destroyOnDrop yields true.

Now, I'm not sure why that code block is needed, I'm guessing there's something specific to mech cluster turret ammo that makes controlling tradeability and craftability via XML unfeasible. Someone who's more familiar with Royalty content should review this and change as needed to restore sellable mech ammo.

**Comment by N7Huntsman** [comment]
Some of the overlap is likely because at least one of the mech turrets is essentially just a mounted heavy charge blaster and uses the same ammo. 

@zhrocks11 @alicecomma may know more about this, but my hunch is that this was attempted as a way to prevent players from farming mech cluster ammo drop beacons for infinite ammo to sell. Unfortunately, it sort of ends up having the opposite effect of creating a situation where players are stuck with a large amount of wealth they're unable to easily dispose of (short of manually piling it all somewhere and burning it).

While wealth from mech ammo is something that that should be addressed, I don't think making it unsellable is the solution

**Comment by N7Huntsman** [comment]
Resolved by #38 .
